tests/test_cnes_sonar_scanner.py

```diff
@@ -0,0 +1,517 @@
+                    f"{cls._PROJECT_ROOT_DIR}/.sonarcache": {'bind': '/opt/sonar-scanner/.sonar/cache', 'mode': 'rw'}
+                })
+            # Wait for SonarQube to process the results
+            time.sleep(8)
+            # Switch back to the Sonar way QP (in case the test needs to be rerun)
+            requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/add_project",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+                data={
+                    "language": language_key,
+                    "project": project_key,
+                    "qualityProfile": "Sonar way"
+                })
+            # Get the new number of issues
+            output = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/issues/search?componentKeys={project_key}",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['issues']
+            issues = [ issue for issue in output if issue['status'] in ('OPEN', 'TO_REVIEW') ]
+            # Hint: if this test fails, there should be {nb_issues_cnes_qp} issues on the {language_name} dummy project with the {cnes_qp} QP but {len(issues)} were found
+            assert len(issues) == nb_issues_cnes_qp
+
+    @classmethod
+    def analysis_tool(cls, tool: str, cmd: str, ref_file: str, tmp_file: str, store_output: bool = True):
+        """
+        This function tests that the image can run a specified code analyzer
+        and that it keeps producing the same result given the same source code.
+
+        :param tool: tool name
+        :param cmd: tool command line
+        :param ref_file: analysis results reference file (path from the root of the project)
+        :param tmp_file: temporary results file (path from the root of the project)
+        :param store_output: (optional) store the standard output in the temporary result file, default: True
+
+        Example (not a doctest):
+            ref = "tests/c_cpp/reference-cppcheck-results.xml"
+            output = "tests/c_cpp/tmp-cppcheck-results.xml"
+            cmd = f"cppcheck --xml-version=2 tests/c_cpp/cppcheck/main.c --output-file={output}"
+            self.analysis_tool("cppcheck", cmd, ref, output, False)
+        """
+        # Run an analysis with the tool
+        docker_client = docker.from_env()
+        output = docker_client.containers.run(cls._SONAR_SCANNER_IMAGE, cmd,
+            auto_remove=True,
+            user=f"{os.getuid()}:{os.getgid()}",
+            volumes={f"{cls._PROJECT_ROOT_DIR}": {'bind': '/usr/src', 'mode': 'rw'}}).decode("utf-8")
+        if store_output:
+            with open(os.path.join(cls._PROJECT_ROOT_DIR, tmp_file), "w", encoding="utf8") as f:
+                f.write(output)
+        # Compare the result of the analysis with the reference
+        # Hint: if this test fails, look for differences with: diff {tmp_file} {ref_file}
+        assert filecmp.cmp(os.path.join(cls._PROJECT_ROOT_DIR, tmp_file), os.path.join(cls._PROJECT_ROOT_DIR, ref_file))
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6252273)